python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -231,6 +249,79 @@ def run(self, splash=None, version=None):
         result = app.exec_()
         return result
 
+    def uses_legacy_authentication(self):
+        """
+        Returns if the Desktop's bootrap code uses the tk-framework-login for
+        logging in.
+
+        :returns: True the bootstrap logic is older than 1.1.0, False otherwise.
+        """
+        return StrictVersion(self.app_version) < StrictVersion("1.1.0")
+
+    def create_legacy_login_instance(self):
+        """
+        Creates the pre-bootstrap-1.1.0 ShotgunLogin used to authenticate.
```

Comment:
Could this be a little more descriptive? Just to help future readers of this code. When we re-read this code 8 months from now and have long forgotten about the security work, a slightly more verbose comment would probably be useful just to summarize. The current docstring explains the "what" - which is great. But with compatibility stuff like this, also providing a short "why" often helps when going back to the code. Hope it makes sense!